p2p/libp2p/libp2pConnectionMonitor.go

```diff
@@ -40,16 +44,62 @@ func (lcm *libp2pConnectionMonitor) Listen(network.Network, multiaddr.Multiaddr)
 // ListenClose is called when network stops listening on an addr
 func (lcm *libp2pConnectionMonitor) ListenClose(network.Network, multiaddr.Multiaddr) {}
 
+// ThresholdDiscoveryPause if the number of connected peers is over this value, the dht discovery is stopped
+func (lcm *libp2pConnectionMonitor) ThresholdDiscoveryPause() int {
+	if lcm.targetConnCount > 0 {
+		return lcm.targetConnCount
+	}
+	return math.MaxInt32
+}
+
+// ThresholdDiscoveryResume if the number of connected peers drop under this value, the dht discovery is restarted
+func (lcm *libp2pConnectionMonitor) ThresholdDiscoveryResume() int {
+	if lcm.targetConnCount > 0 {
+		return lcm.targetConnCount * 4 / 5
+	}
+	return 0
+}
+
+// ThresholdRandomTrim if the number of connected peers is over this value, we start cutting of connections at random
+func (lcm *libp2pConnectionMonitor) ThresholdRandomTrim() int {
+	if lcm.targetConnCount > 0 {
+		return lcm.targetConnCount * 6 / 5
+	}
+	return math.MaxInt32
+}
+
+// Request a reconnet to initial list
+func (lcm *libp2pConnectionMonitor) doReconn() {
+	select {
+	case lcm.chDoReconnect <- struct{}{}:
+	default:
+	}
+}
+
 // Connected is called when a connection opened
-func (lcm *libp2pConnectionMonitor) Connected(network.Network, network.Conn) {}
+func (lcm *libp2pConnectionMonitor) Connected(netw network.Network, conn network.Conn) {
```

Comment:
Should improve code coverage of the following funcs: Connected, ThresholdRandomTrim, ThresholdDiscoveryResume, ThresholdDiscoveryPause, Disconnected